XamarinCommunityToolkit/Behaviors/BaseBehavior.shared.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+using XamarinCommunityToolkit.Controls;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    /// <summary>
+    /// Contains base functionality for all derived behaviors
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    [Preserve(AllMembers = true)]
+    public class BaseBehavior<T> : Behavior<T> where T : BindableObject
```

Comment:
Does this need to be public? Do we expect other people to work with this as well?

Reply:
> Does this need to be public? Do we expect other people to work with this as well?

Sure, otherwise we will get "inconsistent accessibility" compile error